components/KiboCollectedProduct.vue

```diff
@@ -178,11 +185,39 @@ export default defineComponent({
       await removeCartItem(props.cartItemId)
     }
 
+    // Get Fullfillment Options
+    const handleFulfillmentOption = async (
+      selectedFulfillmentValue?: string,
+      shouldOpenModal?: boolean
+    ) => {
+      const itemToBeUpdated = cart.value.items.find((item) => item.id === props.cartItemId)
+      if (selectedFulfillmentValue === "Ship") {
+        itemToBeUpdated.fulfillmentMethod = selectedFulfillmentValue
+        itemToBeUpdated.purchaseLocation = null
+      }
+
+      if (shouldOpenModal) {
+        modal.show({
+          component: StoreLocatorModal,
+          props: {
+            handleSetStore: async (selectedStore) => {
+              itemToBeUpdated.fulfillmentMethod = "Pickup"
+              itemToBeUpdated.fulfillmentLocationCode = selectedStore
+              itemToBeUpdated.purchaseLocation = selectedStore
+              await updateCartItem(props.cartItemId, itemToBeUpdated)
+            },
+          },
+        })
+      } else {
+        await updateCartItem(props.cartItemId, itemToBeUpdated)
```

Comment:
How is the "itemToBeUpdated" being set to "Pickup" at this else condition?
We should also clone the itemToBeUpdated and data being passed to updateCartItem instead of manipulating the reference pulled from the cart state ref

Reply:
itemToBeUpdated is not being set to Pickup in the else condition as whenever we will try to add one cartItem to pickup, the storeLocatorModal will open.

I have resolved the cloning part.

Reply:
If shouldOpenModal === false, but the user clicked the radio button for "Pickup". I don't see anything in the code that would actually change the cartitem to "Pickup"